core/src/Revolution/modLexicon.php

```diff
@@ -581,13 +583,37 @@ private function _parse($str, $params)
             return $str;
         }
 
+        $params = $this->_flatten($params);
         foreach ($params as $k => $v) {
             $str = str_replace('[[+' . $k . ']]', $v, $str);
         }
 
         return $str;
     }
 
+    /**
+     * Flattens an array by dot notation
+     *
+     * @access private
+     * @param array $array The array to flatten
+     * @return array The processed array
+     */
+    private function _flatten($array) {
```

Comment:
I agree with such kind of solution for 2.x, but for 3.x it should be done as a feature that allows making array a flat. And for this case need to reuse such a helper.

Reply:
@Jako Can you help with this?

Reply:
@alroniks Can you do this in a separate PR? Since I myself will not be able to implement it.

Reply:
@alroniks I think it's enough to remove the underscore from the beginning of the name. Where to move this function? In the MODX class? He was already a God class. You need to put it in a separate service. But it hasn't been created yet.

Reply:
I'm in favor of @sergant210's proposal for now, this is blocking a bug and we can always add a helper function down the road. What do you think @alroniks ?

Reply:
@Ibochkarev Can you please just remove the underscore from the method name at both places?